test/test.cc

```diff
@@ -4302,10 +4307,11 @@ TEST(SSLClientTest, ServerCertificateVerification4) {
   cli.set_connection_timeout(30);
 
   auto res = cli.Get("/test");
-  ASSERT_TRUE(res);
-  ASSERT_EQ(200, res->status);
 
   t.join();
+
```

Comment:
@gjasny, I don't think this change is needed. (Also I see the same change a number of times in this file.) Could you reverted these?

Reply:
It is needed that the test does not crash if the assertion is not fulfilled. In contrast to `EXPECT` the `ASSERT` macros return on error. If a `std::thread` is destructed and it is neither detached nor joined it will throw an assertion and the test will break.

Reply:
If you'd like I could move them into a different PR. But they are already different commits.

Reply:
Yes, could you take the unrelated changes to a separated pull request, so that this pull request becomes simpler and clearer?